File: backend/app/routers/itinerary.py
from fastapi import APIRouter

from app.schemas.itinerary import ItineraryRequest, ItineraryResponse
from app.services import itinerary_service

router = APIRouter(prefix="/api", tags=["itinerary"])


@router.post("/itinerary", response_model=ItineraryResponse)
def build(payload: ItineraryRequest) -> ItineraryResponse:
    return itinerary_service.build_itinerary(
        payload.user_profile, payload.travel_date, payload.selected_place_ids
    )

File: frontend/src/api/itinerary.ts
import { client } from "./client";
import type { UserProfile } from "../types/user";
import type { Itinerary } from "../types/itinerary";

export async function postItinerary(
  user_profile: UserProfile,
  travel_date?: string,
  selected_place_ids?: string[]
): Promise<Itinerary> {
  const { data } = await client.post<Itinerary>("/api/itinerary", {
    user_profile,
    travel_date,
    selected_place_ids: selected_place_ids ?? [],
  });
  return data;
}
